azurelinuxagent/common/osutil/default.py

```diff
@@ -440,14 +442,138 @@ def get_first_if(self):
             logger.warn(('SIOCGIFCONF returned more than {0} up '
                          'network interfaces.'), expected)
         sock = buff.tostring()
+        primary = self.get_primary_interface()
         for i in range(0, struct_size * expected, struct_size):
             iface=sock[i:i+16].split(b'\0', 1)[0]
-            if iface == b'lo' or iface.startswith('lo:'):
+            if len(iface) == 0 or self.is_loopback(iface) or iface != primary:
+                # test the next one
```

Comment:
might be nice to add logs so we can see from the logs what happened in this loop (only if this has a risk of going bad).